core/node/vm_runner/src/process.rs

```diff
@@ -148,39 +177,16 @@ impl VmRunner {
 
             let last_ready_batch = self
                 .io
-                .last_ready_to_be_loaded_batch(&mut self.pool.connection().await?)
+                .last_ready_to_be_loaded_batch(&mut self.pool.connection_tagged("vm_runner").await?)
                 .await?;
             METRICS.last_ready_batch.set(last_ready_batch.0.into());
             if next_batch > last_ready_batch {
                 // Next batch is not ready to be processed yet
                 tokio::time::sleep(SLEEP_INTERVAL).await;
                 continue;
             }
-            let Some((batch_data, storage)) = self.loader.load_batch(next_batch).await? else {
-                // Next batch has not been loaded yet
-                tokio::time::sleep(SLEEP_INTERVAL).await;
-                continue;
-            };
-            let batch_executor = self.batch_executor_factory.init_batch(
-                storage,
-                batch_data.l1_batch_env.clone(),
-                batch_data.system_env.clone(),
-            );
-            let output_handler = self
-                .output_handler_factory
-                .create_handler(batch_data.system_env, batch_data.l1_batch_env)
-                .await?;
-
-            self.io
-                .mark_l1_batch_as_processing(&mut self.pool.connection().await?, next_batch)
-                .await?;
-            let handle = tokio::task::spawn(Self::process_batch(
-                batch_executor,
-                batch_data.l2_blocks,
-                output_handler,
-            ));
+            let handle = tokio::spawn(self.clone().process_batch(next_batch));
```

Comment:
I'm not 100% sure this change is correct for existing VM runner components other than the VM playground. With this async pipeline, it means that `mark_l1_batch_as_processing()` can be called out of order (e.g., first for batch 5, then 3, then 4).

Reply:
The order of `mark_l1_batch_as_processing` shouldn't matter as long as `mark_l1_batch_as_completed` is sequential. All existing implementations use processing stage purely for metrics/observability. Maybe we should reflect that the implementation should not rely on order of execution in `mark_l1_batch_as_processing`'s rustdoc

Reply:
Ah, sorry, I skimmed through the code somehow missed that